queue.go

```diff
@@ -94,18 +117,20 @@ type Queue struct {
 }
 
 // NewQueue creates and returns a new Queue structure
-func NewQueue(url string, name string, prefetchSize int, isConsumer, durable bool, jobs chan amqp.Delivery) (*Queue, error) {
+func NewQueue(name string, prefetchSize int, isConsumer, durable bool, jobs chan amqp.Delivery) (*Queue, error) {
+	if cfg.HostName == "" || cfg.Port == "" ||
```

Comment:
Validation here is too late because if the variables are missing client services will run fine but throw an error later. Validation has to be done when reading the variables

Reply:
done